pkg/util/helper/workstatus.go

```diff
@@ -84,22 +85,34 @@ func assembleWorkStatus(c client.Client, selector labels.Selector, workload *uns
 			klog.Errorf("Failed to get manifestIndex of workload in work.Spec.Workload.Manifests. Error: %v.", err)
 			return nil, err
 		}
+		clusterName, err := names.GetClusterName(work.Namespace)
+		if err != nil {
+			klog.Errorf("Failed to get clusterName from work namespace %s. Error: %v.", work.Namespace, err)
+			return nil, err
+		}
+
+		// if sync work to member cluster failed, then set status back to resource binding.
+		for _, c := range work.Status.Conditions {
+			if c.Status == v1.ConditionFalse {
+				aggregatedStatus := workv1alpha1.AggregatedStatusItem{
+					ClusterName:    clusterName,
+					Applied:        false,
+					AppliedMessage: c.Message,
+				}
+				statuses = append(statuses, aggregatedStatus)
```

Comment:
This will append an `extra items, right?

Reply:
Just draft one
```diff
diff --git a/pkg/util/helper/workstatus.go b/pkg/util/helper/workstatus.go
index dfdd746d..74c1a298 100644
--- a/pkg/util/helper/workstatus.go
+++ b/pkg/util/helper/workstatus.go
@@ -6,7 +6,7 @@ import (
        "reflect"

        "k8s.io/apimachinery/pkg/api/meta"
-       v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+       metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
        "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
        "k8s.io/apimachinery/pkg/labels"
        "k8s.io/apimachinery/pkg/runtime/schema"
@@ -103,6 +103,22 @@ func assembleWorkStatus(c client.Client, selector labels.Selector, workload *uns
                        }
                }

+               var applied bool
+               var appliedMsg string
+               if cond := meta.FindStatusCondition(work.Status.Conditions, workv1alpha1.WorkApplied); cond != nil {
+                       switch cond.Status {
+                       case metav1.ConditionTrue:
+                               applied = true
+                       case metav1.ConditionUnknown:
+                               fallthrough
+                       case metav1.ConditionFalse:
+                               applied = false
+                               appliedMsg = cond.Message
+                       default: // should not happen unless the condition api changed.
+                               panic("unexpected status")
+                       }
+               }
+
                for _, manifestStatus := range work.Status.ManifestStatuses {
                        equal, err := equalIdentifier(&manifestStatus.Identifier, identifierIndex, workload)
                        if err != nil {
@@ -110,9 +126,10 @@ func assembleWorkStatus(c client.Client, selector labels.Selector, workload *uns
                        }
                        if equal {
                                aggregatedStatus := workv1alpha1.AggregatedStatusItem{
-                                       ClusterName: clusterName,
-                                       Status:      manifestStatus.Status,
-                                       Applied:     true,
+                                       ClusterName:    clusterName,
+                                       Status:         manifestStatus.Status,
+                                       Applied:        applied,
+                                       AppliedMessage: appliedMsg,
                                }
                                statuses = append(statuses, aggregatedStatus)
                                break
```

Reply:
actually, if failed to sync workload to member cluster, there would be no `ManifestStatuses` in work object.